app/services/triannon/ldp_writer.rb

```diff
@@ -387,9 +388,9 @@ def create_resources_in_container(predicate)
         }
 
         if (predicate == RDF::Vocab::OA.hasTarget)
-          resource_ids << create_resource(graph_for_resource.to_ttl, "#{@id}/t")
+          resource_ids << create_resource(graph_for_resource.to_ttl, "#{@root_container}/#{@id}/t")
```

Comment:
If you have the annotation object around, this seems like another consumer for a method that does this concatenation for you.